### ISSUE TITLE
Warning (unminified build) failing silently in Firefox + Firebug

### ISSUE THREAD

**Issue by fabd** [vuejs/vue#4454]
Hi folks,

As a new Vue developer I bumped into an issue where Vue warnings do not show and fail silently under Firefox.+ Firebug. 

I can detail this as best I can if necessary, but I'm wondering if you've heard of this issue already?

As you can imagine, this was quite frustrating because I spent a while trying to understand what doesn't work ... only a couple days later I try the page in Google Chrome and I finally see the hints that Vue was giving all along.

### What works

Vue gives a warning when the v-on:click tries to do "show = false" in a simple exercise (cf. scenario below):

    vue.js:513 [Vue warn]: Avoid mutating a prop directly since the value (...)

### What doesn't work

In Firefox 50.0.2 (**Ubuntu VM**) + Firebug 2.0.18 (addon), nothing shows up, _it fails silently_.

If I enable *Firebug > Console > Show Chrome Messages* in the same scenario, I then see:

    TypeError: dbg is undefined    ....... 	     debuggerLib.js (line 556, col 9)

In Firefox 50.0.2 (**Windows version**) + *builtin* Firebug, the Vue warning does show up.

I don't know what happens with the addon version of Firebug + Firefox 50.0.2 + Windows since I accepted Firebug to switch to the new "builtin" version, I don't know how to revert it to the addon (which I prefer).

### Other misc

My page includes **VueJS 2.1.4 development build** [downloaded here](https://github.com/vuejs/vue/blob/dev/dist/vue.js) on December 2. Right click file size info: "213,3 kB (213.319 bytes)". I downloaded also 2.1.4 (unminified) from a CDN (don't remember which). Same file size, same results.

### The scenario

... is me basically trying to make vue-strap Alert component work in Vue 2 (as an exercise).  I took [the code here](https://github.com/yuche/vue-strap/blob/master/src/Alert.vue) and made it a simple one page. Mind you this Alert in Vue 2 is broken in other ways (I think, `<transition>` is needed, I had to remove "coerce" etc.).. but it sounds like the issue here is related to the development environment (Firebug) since it does not happen in Google Chrome.

**Comment by posva** [comment]
> I don't know what happens with the addon version of Firebug + Firefox 50.0.2 + Windows since I accepted Firebug to switch to the new "builtin" version, I don't know how to revert it to the addon (which I prefer)

It's probably a firebug bug then. You should open an issue on their repo.

Try to reinstall firebug to revert the changes.

**Comment by markbrown4** [comment]
I ran into this too, the app started silently failing and no errors were visible in Firefox.

Loading in Chrome worked and showed the cause:

> [Vue warn]: Unknown custom element: <drop-down>

Assuming you want to make Vue work in more than just Chrome, you'll need to make these warnings not cause Firefox to fall over completely.

**Comment by fabd** [comment]
@markbrown4  I found the same version of Firefox under Windows 7 x64 worked. It is annoying, it seems related to either a) running Firefox inside a VM (doubtful) 2) something to do with the Ubuntu version  3) something to do with the way they handled the transition from Firebug to their builtin firebug merge. At first I turned down their offer to use the builtin dev tools, and then later I think I accepted. Then I realized my mistake because even with the firebug theme it doesn't look the same and it's really sluggish (which is ironic since it's supposed to be more performant as a "native" dev tool). But, I don't know how to revert it.

Long story short, I installed Chromium as my dev browser, and use Google Chrome as my leisure browser. I didn't find a resolution for this issue. I also tried a Firefox "ESR" release I think but it was already displaying CSS issues  (something with transform origin), so I had to scrap that.

**Comment by markbrown4** [comment]
I'm running latest versions of Firefox and Firebug on Mac and see the issue, seems like a legit bug to me.

**Comment by posva** [comment]
@markbrown4 it is, but a Firebug one 🙂

**Comment by markbrown4** [comment]
@posva I just don't buy that.

Asking everyone who wants to use your software to develop in Chrome is not a good enough answer.

**Comment by posva** [comment]
> Asking everyone who wants to use your software to develop in Chrome is not a good enough answer.

We don't do that. We recommend Chrome to benefit from the Vue devtools and enjoy a better DX

It's a Firebug bug, unrelated to Vue.js. If someone ever finds a way of fixing it, we'll happily welcome a PR! I'm running the latest version of Firefox on mac too and cannot see the issue. If you're still running the Firebug addon, you should know it's not maintained anymore and that Firefox devtools integrate them (http://getfirebug.com/).
You can find more issues around: https://github.com/firebug/firebug/issues/8062

**Comment by fabd** [comment]
> You can find more issues around

Very interesting, thanks for confirming.

**Comment by markbrown4** [comment]
Yeah, interesting one.  Firebug does seem to have broken `console.error()` here completely.

> console.error('test')
> TypeError: dbg is undefined

It's fair enough that you don't support unmaintained plugins, it's time for me to move on from it.
However if you did want to keep Vue working in these cases you could use `console.warn` for these, they are technically warnings.
